mlrun/frameworks/langchain/model_server.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# flake8: noqa  - this is until we take care of the F401 violations with respect to __all__ & sphinx
+
+from typing import Any, Dict, Union
+
+import langchain_community.llms
+from langchain_core.language_models.llms import LLM
+
+import mlrun
+from mlrun.serving.v2_serving import V2ModelServer
+
+
+class LangChainModelServer(V2ModelServer):
+    """
+    LangChain Model serving class, inheriting the V2ModelServer class for being initialized automatically by the model
+    server and be able to run locally as part of a nuclio serverless function, or as part of a real-time pipeline.
+    For compatibility with langchain models, this class supports the following operations:
+    - invoke
+    - batch
+    - ainvoke
+    - abatch
+    in the future we will add streaming support.
+    Notice: In order to use this serving class, please ensure that the langchain_community package is installed.
+    """
+
+    def __init__(
+        self,
+        context: mlrun.MLClientCtx = None,
+        model_class: str = None,
+        llm: Union[str, LLM] = None,
```

Comment:
I see `llm` can be a string as well, what would be the use case? Doesn't it replace the `model_class` in that case? If so, delete the `model_class`.